test/extensions_testsuite/reduce_async_sycl.pass.cpp

```diff
@@ -0,0 +1,116 @@
+// -*- C++ -*-
```

Comment:
The test should be introduced in test/parallel_api/experimental or test/parallel_api/*/async to be consistent with the refactoring being done in #81